v2/internal/frontend/desktop/darwin/frontend.go

```diff
@@ -352,7 +350,9 @@ func (f *Frontend) processRequest(r *request) {
 		headersLen = len(headerData)
 	}
 
-	C.ProcessURLResponse(r.ctx, r.url, C.int(rw.Code), headers, C.int(headersLen), content, C.int(contentLen))
+	url := C.CString(r.url)
+	defer C.free(unsafe.Pointer(url))
```

Comment:
Wondering if we should free this in the C function. Should be fine as is as it's the same thread but just wondering...

Reply:
From the perspective of responsibilities I would expect the free here, because the memory has been allocated here.

Normally as a callee I would not expect it's needed to free any incoming pointers, because I would expect this is done by the caller which allocated it.
If the data must life beyond the lifetime of a method call, it should be the responsibility of the callee to copy the data or clearly state that the callee takes the ownership of the allocated memory.

Reply:
The first thing `ProcessURLResponse` does is copy the url into an `NSString` which allows for reference counting on the Objective-C side.